TP/steps/19-form-validation-solution/src/app/shared/form/form.component.html

```diff
@@ -12,91 +10,112 @@
+              <span *ngIf="lastname.touched && lastname.errors?.required"><br />Last name is required</span>
               <span *ngIf="lastname.dirty && lastname.errors?.minlength"><br />Last name is 2 chars min</span>
             </p>
           </td>
           <td class="align-right">
-            <img width="100" src="{{ model.photo || 'https://randomuser.me/api/portraits/lego/6.jpg' }}" />
-            <input type="hidden" [(ngModel)]="model.photo" name="photo" />
+            <img alt="photo" width="100" src="{{ model?.photo || 'https://randomuser.me/api/portraits/lego/6.jpg' }}" />
+            <input type="hidden" [ngModel]="model?.photo" name="photo" />
           </td>
         </tr>
       </table>
 
-      <p [class.errors]="email.dirty && email.invalid">
+      <p [class.errors]="email.touched && email.invalid">
         <mat-form-field>
           <input
-            [(ngModel)]="model.email"
+            [ngModel]="model?.email"
+            (ngModelChange)="model.email = $event"
             matInput
-            placeholder="Email"
-            required
+            placeholder="Email *"
+            style="width:100%"
             name="email"
+            required
             type="email"
-            style="width:100%"
             #email="ngModel"
           />
         </mat-form-field>
-
-        <span *ngIf="email.dirty && email.errors?.required">Email is mandatory</span>
+        <span *ngIf="email.touched && email.errors?.required">Email is mandatory</span>
       </p>
 
       <p ngModelGroup="address">
         <mat-form-field>
-          <input [(ngModel)]="model.address.street" matInput placeholder="Address" style="width:100%" name="street" />
+          <input
+            [ngModel]="model?.address?.street"
+            (ngModelChange)="model.address.street = $event"
+            matInput
+            placeholder="Address"
+            style="width:100%"
+            name="street"
+          />
         </mat-form-field>
       </p>
 
       <p ngModelGroup="address">
         <mat-form-field>
-          <input [(ngModel)]="model.address.city" matInput placeholder="City" name="city" />
+          <input
+            [ngModel]="model?.address?.city"
+            (ngModelChange)="model.address.city = $event"
+            matInput
+            placeholder="City"
+            name="city"
+          />
         </mat-form-field>
         <mat-form-field>
-          <input [(ngModel)]="model.address.postalCode" matInput placeholder="Postal Code" name="postalCode" />
+          <input
+            [ngModel]="model?.address?.postalCode"
+            (ngModelChange)="model.address.postalCode = $event"
+            matInput
+            placeholder="Postal Code"
+            name="postalCode"
+          />
         </mat-form-field>
       </p>
 
       <p [class.errors]="phone.dirty && phone.invalid">
         <mat-form-field>
           <input
-            [(ngModel)]="model.phone"
+            [ngModel]="model?.phone"
```

Comment:
question: what is the problem with previous syntax ?

Reply:
Well, I was trying to avoid errors since the model is undefined before the Input is set (after adding the People type). But I think it would be better to keep the `[(ngModel)]` syntax and just set a default value to the model in the `ngOnInit` hook. I will refactor the exercises in that way if it's ok for you ?